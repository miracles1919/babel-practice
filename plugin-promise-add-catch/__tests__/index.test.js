const fs = require('fs');
const path = require('path');
const pluginTester = require('babel-plugin-tester').default;

const PromiseAddCatchPlugin = require('../plugin/index');

const source = fs
  .readFileSync(path.join(__dirname, '../source2.js'))
  .toString();

pluginTester({
  plugin: PromiseAddCatchPlugin,
  pluginName: 'PromiseAddCatchPlugin',
  babelOptions: {
    parserOpts: {
      sourceType: 'unambiguous',
    },
  },
  tests: {
    'names': {
      code: source,
      snapshot: true,
      pluginOptions: {
        names: ['confirm'],
      },
    },
    catchTemplate: {
      code: source,
      snapshot: true,
      pluginOptions: {
        names: ['confirm', 'p'],
        catchTemplate: `console.log(err)`,
      },
    },
    disabledComments: {
      code: source,
      snapshot: true,
      pluginOptions: {
        names: ['confirm', 'p'],
        disabledComments: 'catch-disabled',
      },
    },
  },
});
