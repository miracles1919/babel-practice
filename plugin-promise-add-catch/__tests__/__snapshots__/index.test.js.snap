// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromiseAddCatchPlugin catchTemplate: catchTemplate 1`] = `

/* catch-disabled */
Dialog.confirm()
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  });

const confirm = () =>
  new Promise((resolve) => {
    setTimeout(() => resolve(true), 300);
  });

/* disabled */
confirm().then();

/* pac-disabled */
Dialog['confirm']().then();

const p = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});

p.then();

const p2 = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});

p2.then();

const handleClick = async () => {
  const res = await Dialog.confirm();

  /* catch-disabled */
  const res2 = await Dialog.confirm();
};

async function handleClick2() {
  const res = await Dialog.confirm();

  /* catch-disabled */
  const res2 = await Dialog.confirm();
}

      ↓ ↓ ↓ ↓ ↓ ↓

/* catch-disabled */
Dialog.confirm()
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .catch((err) => {
    console.log(err);
  });

const confirm = () =>
  new Promise((resolve) => {
    setTimeout(() => resolve(true), 300);
  });
/* disabled */

confirm()
  .then()
  .catch((err) => {
    console.log(err);
  });
/* pac-disabled */

Dialog["confirm"]().then();
const p = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});
p.then().catch((err) => {
  console.log(err);
});
const p2 = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});
p2.then();

const handleClick = async () => {
  const res = await Dialog.confirm().catch((err) => {
    console.log(err);
  });
  /* catch-disabled */

  const res2 = await Dialog.confirm().catch((err) => {
    console.log(err);
  });
};

async function handleClick2() {
  const res = await Dialog.confirm().catch((err) => {
    console.log(err);
  });
  /* catch-disabled */

  const res2 = await Dialog.confirm().catch((err) => {
    console.log(err);
  });
}


`;

exports[`PromiseAddCatchPlugin disabledComments: disabledComments 1`] = `

/* catch-disabled */
Dialog.confirm()
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  });

const confirm = () =>
  new Promise((resolve) => {
    setTimeout(() => resolve(true), 300);
  });

/* disabled */
confirm().then();

/* pac-disabled */
Dialog['confirm']().then();

const p = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});

p.then();

const p2 = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});

p2.then();

const handleClick = async () => {
  const res = await Dialog.confirm();

  /* catch-disabled */
  const res2 = await Dialog.confirm();
};

async function handleClick2() {
  const res = await Dialog.confirm();

  /* catch-disabled */
  const res2 = await Dialog.confirm();
}

      ↓ ↓ ↓ ↓ ↓ ↓

/* catch-disabled */
Dialog.confirm()
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  });

const confirm = () =>
  new Promise((resolve) => {
    setTimeout(() => resolve(true), 300);
  });
/* disabled */

confirm()
  .then()
  .catch((err) => err);
/* pac-disabled */

Dialog["confirm"]()
  .then()
  .catch((err) => err);
const p = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});
p.then().catch((err) => err);
const p2 = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});
p2.then();

const handleClick = async () => {
  const res = await Dialog.confirm().catch((err) => err);
  /* catch-disabled */

  const res2 = await Dialog.confirm();
};

async function handleClick2() {
  const res = await Dialog.confirm().catch((err) => err);
  /* catch-disabled */

  const res2 = await Dialog.confirm();
}


`;

exports[`PromiseAddCatchPlugin names: confirm: names: confirm 1`] = `

/* catch-disabled */
Dialog.confirm()
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  });

const confirm = () =>
  new Promise((resolve) => {
    setTimeout(() => resolve(true), 300);
  });

/* disabled */
confirm().then();

/* pac-disabled */
Dialog['confirm']().then();

const p = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});

p.then();

const p2 = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});

p2.then();

const handleClick = async () => {
  const res = await Dialog.confirm();

  /* catch-disabled */
  const res2 = await Dialog.confirm();
};

async function handleClick2() {
  const res = await Dialog.confirm();

  /* catch-disabled */
  const res2 = await Dialog.confirm();
}

      ↓ ↓ ↓ ↓ ↓ ↓

/* catch-disabled */
Dialog.confirm()
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .catch((err) => err);

const confirm = () =>
  new Promise((resolve) => {
    setTimeout(() => resolve(true), 300);
  });
/* disabled */

confirm()
  .then()
  .catch((err) => err);
/* pac-disabled */

Dialog["confirm"]().then();
const p = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});
p.then();
const p2 = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});
p2.then();

const handleClick = async () => {
  const res = await Dialog.confirm().catch((err) => err);
  /* catch-disabled */

  const res2 = await Dialog.confirm().catch((err) => err);
};

async function handleClick2() {
  const res = await Dialog.confirm().catch((err) => err);
  /* catch-disabled */

  const res2 = await Dialog.confirm().catch((err) => err);
}


`;

exports[`PromiseAddCatchPlugin names: names 1`] = `

/* catch-disabled */
Dialog.confirm()
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  });

const confirm = () =>
  new Promise((resolve) => {
    setTimeout(() => resolve(true), 300);
  });

/* disabled */
confirm().then();

/* pac-disabled */
Dialog['confirm']().then();

const p = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});

p.then();

const p2 = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});

p2.then();

const handleClick = async () => {
  const res = await Dialog.confirm();

  /* catch-disabled */
  const res2 = await Dialog.confirm();
};

async function handleClick2() {
  const res = await Dialog.confirm();

  /* catch-disabled */
  const res2 = await Dialog.confirm();
}

      ↓ ↓ ↓ ↓ ↓ ↓

/* catch-disabled */
Dialog.confirm()
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .then(() => {
    console.log(111);
  })
  .catch((err) => err);

const confirm = () =>
  new Promise((resolve) => {
    setTimeout(() => resolve(true), 300);
  });
/* disabled */

confirm()
  .then()
  .catch((err) => err);
/* pac-disabled */

Dialog["confirm"]().then();
const p = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});
p.then();
const p2 = new Promise((resolve) => {
  setTimeout(() => resolve(true), 300);
});
p2.then();

const handleClick = async () => {
  const res = await Dialog.confirm().catch((err) => err);
  /* catch-disabled */

  const res2 = await Dialog.confirm().catch((err) => err);
};

async function handleClick2() {
  const res = await Dialog.confirm().catch((err) => err);
  /* catch-disabled */

  const res2 = await Dialog.confirm().catch((err) => err);
}


`;
